File: scripts/migrate_data.js
const fs = require('fs');
const path = require('path');

const dataTsPath = path.join(__dirname, '..', 'lib', 'data.ts');
let content = fs.readFileSync(dataTsPath, 'utf8');

// 1. Replace "has_india_pricing" to "global_availability"
content = content.replace(/has_india_pricing/g, 'global_availability');

// 2. Replace "inr_price" to "usd_price"
content = content.replace(/inr_price/g, 'usd_price');

// 3. Replace India related words in descriptions and names
// Being somewhat careful to change "Indian businesses" -> "global businesses"
content = content.replace(/India's/g, "The world's");
content = content.replace(/Indian context/g, "global context");
content = content.replace(/Indian businesses/g, "global businesses");
content = content.replace(/Bharat/g, "Global Enterprises");
content = content.replace(/in India/g, "worldwide");
content = content.replace(/for India/g, "globally");
content = content.replace(/Indian dialects/g, "global dialects");
content = content.replace(/Indian languages/g, "multiple languages");
content = content.replace(/Indian/g, "global");

// 4. Price conversion
// We use a regex to find "â‚¹" followed by a number (maybe with commas)
const priceRegex = /â‚¹([0-9,]+)/g;

content = content.replace(priceRegex, (match, p1) => {
  // Removing commas
  const amountStr = p1.replace(/,/g, '');
  const inrAmount = parseInt(amountStr, 10);
  
  if (isNaN(inrAmount)) return match;
  
  // Approximate conversion: 1 USD = ~83 INR, let's round
  let usdAmount = Math.round(inrAmount / 83);
  
  // If it's a very large number, maybe round nicer (e.g. 1000s)
  if (usdAmount > 1000) {
     usdAmount = Math.round(usdAmount / 100) * 100;
  } else if (usdAmount > 100) {
     usdAmount = Math.round(usdAmount / 10) * 10;
  } else if (usdAmount > 50) {
     usdAmount = Math.round(usdAmount / 5) * 5;
  }
  
  if (usdAmount === 0 && inrAmount > 0) usdAmount = 1;
  
  return '$' + usdAmount.toLocaleString('en-US');
});

// Since the seed data arrays still have hardcoded strings like "â‚¹2,999/month", they will be fixed naturally by the Regex.
// Same for the "inr_price": "â‚¹XX" lines which got renamed to "usd_price": "â‚¹XX" then converted to "$XX".

fs.writeFileSync(dataTsPath, content);
console.log('Successfully globalized lib/data.ts');

File: scripts/migrate_data.py
import re
import os

data_ts_path = os.path.join(os.path.dirname(__file__), '..', 'lib', 'data.ts')

with open(data_ts_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Replace "has_india_pricing" to "global_availability"
content = content.replace("has_india_pricing", "global_availability")

# 2. Replace "inr_price" to "usd_price"
content = content.replace("inr_price", "usd_price")

# 3. Replace India related words down
content = content.replace("India's", "The world's")
content = content.replace("Indian context", "global context")
content = content.replace("Indian businesses", "global businesses")
content = content.replace("Bharat", "Global Enterprises")
content = content.replace("in India", "worldwide")
content = content.replace("for India", "globally")
content = content.replace("Indian dialects", "global dialects")
content = content.replace("Indian languages", "multiple languages")
content = content.replace("Indian", "global")

# 4. Price conversion
def convert_price(match):
    amount_str = match.group(1).replace(',', '')
    try:
        inr_amount = int(amount_str)
        usd_amount = round(inr_amount / 83.0)
        
        if usd_amount > 1000:
            usd_amount = round(usd_amount / 100) * 100
        elif usd_amount > 100:
            usd_amount = round(usd_amount / 10) * 10
        elif usd_amount > 50:
            usd_amount = round(usd_amount / 5) * 5
            
        if usd_amount == 0 and inr_amount > 0:
            usd_amount = 1
            
        return '$' + f"{usd_amount:,}"
    except ValueError:
        return match.group(0)

# match "₹" followed by digits and commas
content = re.sub(r'₹([0-9,]+)', convert_price, content)

with open(data_ts_path, 'w', encoding='utf-8') as f:
    f.write(content)

print('Successfully globalized lib/data.ts using Python')
